app/ts/components/DebounceInput.tsx

```diff
@@ -0,0 +1,59 @@
+import { Signal } from '@preact/signals'
+import { useRef, useState } from 'preact/hooks'
+
+type DebounceInputProps = {
+	placeholder?: string
+	delay?: number
+	onDebouncedChange: (value: string) => void
+	onValidationError: (value: string) => void
+	validate: (value: string) => boolean
+	validationError?: string
+	inputValue: Signal<string>
+}
+
+export function DebounceInput({
+	placeholder = '',
+	delay = 500,
+	onDebouncedChange,
+	onValidationError,
+	validate,
+	validationError = 'Invalid input.',
+	inputValue,
+}: DebounceInputProps) {
+	const inputTimeoutRef = useRef<number | null>(null)
+	const [error, setError] = useState<string | undefined>(undefined)
```

Comment:
Consider using a Signal instead of state.  In most cases they are superior.

Reply:
added issue